repair/row_level.cc

```diff
@@ -832,10 +832,15 @@ class repair_meta {
     }
 
     // Get a list of row hashes in _working_row_buf
-    std::unordered_set<repair_hash>
+    future<std::unordered_set<repair_hash>>
     working_row_hashes() {
-        return boost::copy_range<std::unordered_set<repair_hash>>(_working_row_buf |
-                boost::adaptors::transformed([] (repair_row& r) { return r.hash(); }));
+        return do_with(std::unordered_set<repair_hash>(), [this] (std::unordered_set<repair_hash>& hashes) {
+            return do_for_each(_working_row_buf, [&hashes] (repair_row& r) {
+                hashes.emplace(r.hash());
```

Comment:
Note, it can still cause a rare stall when enlarging the hash table. We'll need to work on a non-stalling hash table (will be useful in other places too).

Reply:
I created https://github.com/scylladb/scylla/issues/4688 to track this issue.